exporter/stackdriver/stackdriver.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package stackdriver contains the OpenCensus exporters for
+// Stackdriver Monitoring and Tracing.
+//
+// Please note that the Stackdriver exporter is currently experimental.
+//
+// Create an exporter:
+//
+// 	import (
+// 		"go.opencensus.io/exporter/stackdriver"
+// 		"go.opencensus.io/trace"
+// 		"go.opencensus.io/stats"
+// 	)
+//
+// 	exporter, err := stackdriver.NewExporter(stackdriver.Options{ProjectID: "google-project-id"})
+// 	if err != nil {
+// 		log.Fatal(err)
+// 	}
+//
+// Export to Stackdriver Trace:
+//
+// 	stats.RegisterExporter(exporter)
+//
+// Export to Stackdriver Trace:
+//
+// 	trace.RegisterExporter(exporter)
+//
+// The package uses Application Default Credentials to authenticate.  See
+// https://developers.google.com/identity/protocols/application-default-credentials
+//
+// Exporter can be used to propagate traces over HTTP requests for Stackdriver Trace.
+//
+// Example:
+//
+//	import "go.opencensus.io/plugin/http/httptrace"
+//
+//	client := &http.Client {
+//		Transport: httptrace.NewTransport(nil, exporter),
+//	}
+//
+// All outgoing requests from client will include a Stackdriver Trace header.
+// See the httptrace package for how to handle incoming requests.
+package stackdriver
+
+import (
+	"net/http"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/trace"
+	"google.golang.org/api/option"
+	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+)
+
+// Options contains options for configuring the exporter.
+type Options struct {
+	// ProjectID is the identifier of the Stackdriver
+	// project the user is uploading the stats data to.
+	ProjectID string
```

Comment:
Can this default from the environment?

Reply:
We can make this configurable via http://cloud.google.com/go/compute/metadata if the user is running their code on GCP.